WordPress/Classes/ViewRelated/Site Creation/SiteIntent/SiteIntentViewController.swift

```diff
@@ -62,10 +62,6 @@ class SiteIntentViewController: CollapsableHeaderViewController {
     }
 
     private func configureCloseButton() {
-        guard navigationController?.viewControllers.first == self else {
-            return
-        }
-
```

Comment:
I'm proposing removing this because:
- it prevents the Cancel button of `SiteIntent/SiteIntentViewController.swift` from being unit tested unless the screen is embedded in a navigation controller
- I'm not aware of why this is needed because from what I can tell, if this screen is present, this screen is always the first screen in the Site Creation flow (please correct me if I'm wrong)

I would love to hear thoughts on this.

Reply:
👋 @guarani - this was added in #18166 in two spots: [Site Intent VC](https://github.com/wordpress-mobile/WordPress-iOS/pull/18166/files#diff-50dff85be9aa08453ec1a4188ab862dd3432813a99fd5778afc4b93326c2de47R65-R67), [Site Design VC](https://github.com/wordpress-mobile/WordPress-iOS/pull/18166/files#diff-77c003d6157397ec7d223ebaa9577d565c5b2546f8a3e9a8b2ac9e855698c99cR114-R116). The thinking behind it was to create a common pattern to only add the "Cancel" button if the VC was the first in the stack of steps, but I didn't think about how it might affect testing.

> from what I can tell, if this screen is present, this screen is always the first screen in the Site Creation flow

That's correct, so it can be removed. It was just a pattern I added to safeguard against VC ordering changing. It (or a similar functionality) should remain in the Site Design VC since it can be first or second, depending on the Site Intent A/B variant.

Reply:
Thanks for clarifying @twstokes, I now better understand why it was added. I would have probably added it too since it makes the screens more robust to changes in the site creation flow (without realizing how it affects testing).